### ISSUE TITLE
Avoid making copies

### ISSUE THREAD

**Issue by gasagna** [The-ReSolver/Fields.jl#7]
This will create copies. Use indexing with `@inbounds` annotations. Loops are fast in Julia, do not fear them.

https://github.com/The-ReSolver/Fields.jl/blob/4ed3216913dbef96dd81c8ac25781d784f0980ba/src/derivatives.jl#L79

**Comment by gasagna** [comment]
Try benchmarking the two and see what's best.

**Comment by tb6g16** [comment]
Explicitly using the loops required with @inbounds is substantially slower than using @views with LinearAlgebra.mul!().

What I've pushed ensures there's no memory allocations in the function, and as far as I can tell the fastest implementation of the functions. The y-derivatives are over an order of magnitude slower than the derivatives, which isn't surprising given the dense nature of the Chebyshev differentiation matrix used to test it.

Lastly I noticed that the indexing was not done in a way that respects the ordering of the elements in memory, so I swapped the order of the nested loops which should have fixed it. This may be problem elsewhere in the code so I will keep an eye out for it.
